tests/system_metrics/test_system_metrics_logging.py

```diff
@@ -59,20 +60,23 @@ def test_manual_system_metrics_monitor():
     assert metrics_history[-1].step > 0
 
 
-def test_automatic_system_metrics_monitor():
+@pytest.mark.parametrize(
+    ("wait_time", "exit_wait_time", "polling_freq"), [(5, 3, 0.5) if is_windows() else (2, 2, 0.2)]
+)
+def test_automatic_system_metrics_monitor(wait_time, exit_wait_time, polling_freq):
```

Comment:
```suggestion
@pytest.mark.parametrize("x", range(1000))
def test_automatic_system_metrics_monitor(x, wait_time, exit_wait_time, polling_freq):
```

to run this test 1000 times to make sure it's not flaky.

Reply:
We can remove this once we confirm the test is not flaky.